Java/core/src/main/java/com/amazon/randomcutforest/store/SmallIndexManager.java

```diff
@@ -106,16 +157,22 @@ public short size() {
     }
 
     /**
-     * Take an index from the free index stack.
+     * Take an index from the free index stack. Note that the stack can be implicit.
```

Comment:
I think notes about how free indexes are computed should be inline comments instead of method documentation. The person calling this method shouldn't have to worry about how the next free index is obtained.

Reply:
Done.